test/boost/total_order_check.hh

```diff
@@ -22,23 +22,27 @@
 #pragma once
 
 #include <vector>
+#include <compare>
 #include <boost/test/unit_test.hpp>
 #include <seastar/util/variant_utils.hh>
 #include <variant>
 #include "test/lib/log.hh"
+#include "utils/collection-concepts.hh"
 
 template<typename Comparator, typename... T>
 class total_order_check {
     using element = std::variant<T...>;
     std::vector<std::vector<element>> _data;
     Comparator _cmp;
 private:
-    static int sgn(int x) {
+    static int sgn(std::strong_ordering x) {
         return (x > 0) - (x < 0);
```

Comment:
I guess it's ok (it's preexisting code which you just changed the parameter type), but wouldn't "x <=> 0" work instead of this wierd calculation?

Reply:
No. x <=> returns an std::strong_ordering value.